### DISCUSSION TITLE
What should be the correct behavior for Filesystem PVC size requests after formatting

### DISCUSSION THREAD

**Discussion by jakobmoellerdev** [topolvm/topolvm#936] [Q&A] [answered]
In our product packaging LVMS, we frequently get the question on how much storage is actually available in a PVC. That question cannot be answered easily because it depends on the filesystem overhead that is generated when formatting the block device used.

Here is an example:
Take a pod, consuming a PVC with a specified request of 1Gi Storage with Filesystem XFS

In this instance, the XFS filesystem will use at least 5-10% for its own metadata accounting, so the actual usable (and reported free) space will be more in the line of 800-900Mi. Of course with mount flags this can change even further.

The question for me is now: Is this correct behavior? If not, how should we accomodate for these defaults and how would we calculate a buffer for the filesystems metadata.

**Comment by llamerada-jp** [comment, marked as answer]
The following issue may be a hint.
https://github.com/container-storage-interface/spec/issues/338

CSI spec authors may be aware of this, but it is not in the specification document.
